cpp/examples/fetch_dependencies.cmake

```diff
@@ -11,17 +11,21 @@
 # or implied. See the License for the specific language governing permissions and limitations under
 # the License.
 # =============================================================================
-set(CPM_DOWNLOAD_VERSION v0.35.3)
+
+include(${CMAKE_CURRENT_LIST_DIR}/versions.cmake)
+
+set(CPM_DOWNLOAD_VERSION v0.38.5)
 file(
   DOWNLOAD
   https://github.com/cpm-cmake/CPM.cmake/releases/download/${CPM_DOWNLOAD_VERSION}/get_cpm.cmake
   ${CMAKE_BINARY_DIR}/cmake/get_cpm.cmake
 )
 include(${CMAKE_BINARY_DIR}/cmake/get_cpm.cmake)
 
-set(CUDF_TAG branch-24.06)
+# find or build it via CPM
 CPMFindPackage(
-  NAME cudf GIT_REPOSITORY https://github.com/rapidsai/cudf
+  NAME cudf
+  FIND_PACKAGE_ARGUMENTS "HINTS ${cudf_ROOT}/latest" GIT_REPOSITORY https://github.com/rapidsai/cudf
```

Comment:
Do we still need this `HINTS`? I think that `latest` is a devcontainer-specific name. We probably don't want that.

Reply:
We do need `HINTS` to tell CMake where to `find_package()` libcudf installation. I think we can use both `${cudf_ROOT}:${cudf_ROOT}/latest' as HINTS so it works with devcontainer and otherwise. Though, it would fall back to installing from `GIT_REPOSITORY` if it can't find the package.

Reply:
Okay, this is fine with me.

Reply:
All done!

Reply:
`HINTS` shouldn't be used. `HINTS` are meant to be locations that are deduced by system introspection. What we have here is a guess based on a layout that occurs only under certain situations.  What you want is to pass this in `PATHS`, as that has significantly lower search priority. 


This does raise the followup question of why aren't builds in devcontainers passing a `-Dcudf_ROOT` or `-Dcudf_DIR`  argument when building the examples. IIRC they are already doing things like this for other projects.